src/utils/comparePurchaseUrgency.js

```diff
@@ -0,0 +1,49 @@
+import { Timestamp } from 'firebase/firestore';
+import { getDaysBetweenDates, itemIsOverdue, itemIsInactive } from './dates';
+
+//comparison function to be passed into Array.sort()
+export function comparePurchaseUrgency(a, b) {
+	/*
+	Here we are prioritizing sorting inactive and/or overdue items first.
+	Overdue items are given priority and placed highest on the list.
+	Inactive items are given lower priority and placed lowest on the list.
+	*/
+	if (!itemIsInactive(a) && itemIsInactive(b)) {
+		return -1;
+	} else if (itemIsInactive(a) && !itemIsInactive(b)) {
+		return 1;
+	} else if (itemIsInactive(a) && itemIsInactive(b)) {
+		return a.name.localeCompare(b.name);
+	}
+
+	if (itemIsOverdue(a) && !itemIsOverdue(b)) {
+		return -1;
+	} else if (!itemIsOverdue(a) && itemIsOverdue(b)) {
+		return 1;
+	} else if (itemIsOverdue(a) && itemIsOverdue(b)) {
+		//if both items are overdue, sort alphabetically
+		return a.name.localeCompare(b.name);
+	}
+
+	//Fallback logic to sort active items (neither inactive nor overdue)
+	/*------------------------------------------------------*/
+	const today = Timestamp.now();
+
+	const daysUntilNextPurchaseA = getDaysBetweenDates(
+		today,
+		a.dateNextPurchased,
+	);
+
+	const daysUntilNextPurchaseB = getDaysBetweenDates(
+		today,
+		b.dateNextPurchased,
+	);
+
+	if (daysUntilNextPurchaseA < daysUntilNextPurchaseB) {
+		return -1;
+	} else if (daysUntilNextPurchaseA > daysUntilNextPurchaseB) {
+		return 1;
+	} else if (daysUntilNextPurchaseA === daysUntilNextPurchaseB) {
+		return a.name.localeCompare(b.name);
```

Comment:
This works as is, but firebase has a built-in function to return the data sorted. It's done via a querySnapshot `query(listCollectionRef, orderBy('name)`. I'll add an example here if you want to look into it further. 

`export async function comparePurchaseUrgency(listId) {
	try {
		const listCollectionRef = collection(db, listId);

		const q = query(listCollectionRef, orderBy('name'));

		const querySnapshot = await getDocs(q);

		const sortedData = [];

		querySnapshot.forEach((doc) => {
			const itemData = { id: doc.id, ...doc.data() };

			const daysUntilPurchase = getDaysBetweenDates(
				new Date(),
				itemData.dateNextPurchased.toDate(),
			);

			itemData.daysUntilPurchase = daysUntilPurchase;
			sortedData.push(itemData);
		});

		sortedData.sort((a, b) => {
			return a.daysUntilPurchase - b.daysUntilPurchase;
		});

		return sortedData;
	} catch (e) {
		console.error('Error querying and sorting data:', e);
		return [];
	}
}`